instance/tests/integration/factories/instance.py

```diff
@@ -23,12 +23,31 @@
 # Imports #####################################################################
 
 import uuid
+from random import randint
 
 import factory
 from factory.django import DjangoModelFactory
+from django.conf import settings
 
 from instance.models.mixins.domain_names import generate_internal_lms_domain
 from instance.models.openedx_instance import OpenEdXInstance
+from instance.models.utils import ConsulAgent
+
+
+def _get_unused_id():
+    """
+    Find an unused ID for Consul
+    """
+    while True:
+        try:
+            uid = randint(2 ** 30, 2 ** 31)
```

Comment:
This line should be placed outside the `try-except` statement as context isn't passed to the exception handler when an error happends, yelding:
```
UnboundLocalError: local variable 'uid' referenced before assignment
```

Reply:
That's weird, I couldn't hit that error, could you explain how you triggered that? E.g:
```
In [1]: try: 
   ...:     uid = 1000 
   ...:     assert False 
   ...: except: 
   ...:     print(uid) 
   ...:                                                                         
1000
```